### PR TITLE
Fix #17616: Quarter-tile scenery is misaligned

### LINKED ISSUE #16392: [Bug] Scenery on sloped surface at wrong height
If scenery, e.g. a tree, is placed on a sloped surface, it should not always be placed at the same base height as the surface, but maybe higher.
- If exactly three corners of the surface are elevated, then the tree should be 2 units higher.
- If exactly two adjacent (i.e. not diagonal) corners are elevated, the tree should be 1 unit higher.
- If it is a so-called diagonal surface (two diagonal corners are elevated by one unit and a third one is elevated 2 units), then the tree should be 2 units higher.
- Otherwise, the tree should be at the same base height as the surface.

But this is not always the case:
image
Here, both trees stand on a similar but rotated sloped surface and should be on the same height, but they clearly are not.

In particular, this happens exactly if the corners
- 1+2
- 2+3
- 1+2+3
are raised.

### LINKED ISSUE #17616: All small scenery pieces are offset by 1 pixel in height from where they should be
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.1-49-g3f56a0e (3f56a0e on develop) provided by GitHub

### Describe the issue

Several users have noticed that all small scenery pieces are now lowered by 1 pixel from their original positions, in all rotations. This is most noticeable with pieces that lie flat on the base of a 1/4 tile like so:

image
image

The full tile square here is a large scenery piece, and the 1/4 tile pieces are small scenery. The 1/4 pieces should be flush against the edge of the large tile, but are 1 pixel lower.

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

1. Build a small scenery piece
2. Note position relative to equivalent large scenery pieces
3. 

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by Gymnasiast**
~~This reverts commit ed18ef11. I think it’s better to do it like this, take the pressure off while Sadret and Duncan work on a solution.~~

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Just make the two line change to Paint.SmallScenery.cpp L161-162 adding a +1 to each.

**Comment by Gymnasiast** [comment]
@Sadret Do you agree with that?

The reason I went for reverting is because Sadret and you did not agree on the fix, and because I expect problems to pop up elsewhere. I just want to get the bug out first and take the pressure off.

**Comment by Sadret** [comment]
> Just make the two line change to Paint.SmallScenery.cpp L161-162 adding a +1 to each.
I think this is the way to go. But add a comment about where the +1 comes from.

**Comment by Gymnasiast** [comment]
@Xtreme97-RCT Could you verify that this solves all the problems you encountered?

File: src/openrct2/paint/tile_element/Paint.SmallScenery.cpp [hunk 1]
```diff
@@ -158,8 +158,8 @@ static void PaintSmallSceneryBody(
     else
     {
         uint8_t quadrant = (sceneryElement.GetSceneryQuadrant() + session.CurrentRotation) & 3;
-        offset.x = SceneryQuadrantOffsets[quadrant].x;
-        offset.y = SceneryQuadrantOffsets[quadrant].y;
+        offset.x = SceneryQuadrantOffsets[quadrant].x - 1;
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Paint.SmallScenery.cpp:161]: @Sadret Could you suggest which comment I should add?

Comment (Sadret) [reply src/openrct2/paint/tile_element/Paint.SmallScenery.cpp:161]: // -1 to maintain compatibility with existing CSOs in context of PR #17343

Comment (Sadret) [reply src/openrct2/paint/tile_element/Paint.SmallScenery.cpp:161]: or maybe 
// -1 to maintain compatibility with existing CSOs in context of issue #17616

**Comment by Gymnasiast** [comment]
I have just received word (via @Richard-L) that this seems to solve the error for NE players.

**Review by Sadret — APPROVED** [review]
LGTM

**Review by Broxzier — APPROVED** [review]

**Pull request merged**